data/abilities.ts

```diff
@@ -5266,20 +5260,7 @@ export const Abilities: import('../sim/dex-abilities').AbilityDataTable = {
 	},
 	wanderingspirit: {
 		onDamagingHit(damage, target, source, move) {
-			if (source.getAbility().flags['failskillswap'] || target.volatiles['dynamax']) return;
-
-			if (this.checkMoveMakesContact(move, source, target)) {
-				const targetCanBeSet = this.runEvent('SetAbility', target, source, this.effect, source.ability);
-				if (!targetCanBeSet) return targetCanBeSet;
-				const sourceAbility = source.setAbility('wanderingspirit', target);
-				if (!sourceAbility) return;
-				if (target.isAlly(source)) {
-					this.add('-activate', target, 'Skill Swap', '', '', `[of] ${source}`);
-				} else {
-					this.add('-activate', target, 'ability: Wandering Spirit', this.dex.abilities.get(sourceAbility).name, 'Wandering Spirit', `[of] ${source}`);
-				}
-				target.setAbility(sourceAbility);
-			}
+			if (this.checkMoveMakesContact(move, source, target)) this.skillSwap(source, target);
```

Comment:
might be worth making the function name more generic, i.e. `swapAbilities`?

Reply:
I wanted it to be clear that the function would make the abilities swap using specifically the logic and messages from Skill Swap but ig there isn't really another way that abilities get swapped so maybe a generic name is better.